tests/test_schemas.py

```diff
@@ -127,6 +131,85 @@
+            {
+                "local_filename": "/path/to/build1.log",
+                "dest_filename": "x86_64-build.log",
+            },
+            {
+                "local_filename": "/path/to/dir1/remote-source.tar.gz",
+                "dest_filename": "remote-source.tar.gz",
+            },
+        ]
+    )
+
+    wf_data.image_components = {'x86_64': [{'type': 'rpm', 'name': 'python-docker-py',
+                                            'version': '1.3.1', 'release': '1.fc24',
+                                            'arch': 'noarch',
+                                            'sigmd5': '7c1f60d8cde73e97a45e0c489f4a3b26',
+                                            'signature': None, 'epoch': None},
+                                           {'type': 'rpm', 'name': 'fedora-repos-rawhide',
+                                            'version': '24', 'release': '0.1', 'arch': 'noarch',
+                                            'sigmd5': 'd41df1e059544d906363605d47477e60',
+                                            'signature': None, 'epoch': None},
+                                           {'type': 'rpm', 'name': 'gpg-pubkey-doc',
+                                            'version': '1.0', 'release': '1', 'arch': 'noarch',
+                                            'sigmd5': '00000000000000000000000000000000',
+                                            'signature': None, 'epoch': None}],
+                                'ppc64le': [{'type': 'rpm', 'name': 'python-docker-py',
+                                             'version': '1.3.1', 'release': '1.fc24',
+                                             'arch': 'noarch',
+                                             'sigmd5': '7c1f60d8cde73e97a45e0c489f4a3b26',
+                                             'signature': None, 'epoch': None},
+                                            {'type': 'rpm', 'name': 'fedora-repos-rawhide',
+                                             'version': '24', 'release': '0.1', 'arch': 'noarch',
+                                             'sigmd5': 'd41df1e059544d906363605d47477e60',
+                                             'signature': None, 'epoch': None},
+                                            {'type': 'rpm', 'name': 'gpg-pubkey-doc',
+                                             'version': '1.0', 'release': '1', 'arch': 'noarch',
+                                             'sigmd5': '00000000000000000000000000000000',
+                                             'signature': None, 'epoch': None}],
+                                }
+
+    with TemporaryDirectory() as d:
+        with open(os.path.join(d, 'workflow_data.json'), 'w') as f:
+            json.dump(wf_data.as_dict(), f, cls=WorkflowDataEncoder)
+        with open(os.path.join(d, 'workflow_data.json')) as f:
+            workflow_json = json.load(f)
+
+    return workflow_json
+
+
+RPM_COMPONENTS_WORKFLOW_DATA = get_workflow_data_json()
+
```

Comment:
I might be missing something but... where is the test? :smile: 

1. pytest only treats functions called `test_*` as tests (AFAIK)
2. this function does not use the workflow data schema

Btw there is a test_inner.py::TestWorkflowData test class, which could also be a good place for the rpm components schema unit test (up to you)

Reply:
I forgot to add it to the fixture. Now it's used in the fixture. I added it to the schema test since we're really testing schema in this test and not the `ImageWorkflowData` object itself, which I'm just using to generate the json.